src/models/room.ts

```diff
@@ -1626,16 +1697,16 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
         if (Number.isInteger(invitedCount)) {
             this.currentState.setInvitedMemberCount(invitedCount!);
         }
-        if (Array.isArray(heroes)) {
-            // be cautious about trusting server values,
-            // and make sure heroes doesn't contain our own id
-            // just to be sure
-            this.summaryHeroes = heroes.filter((userId) => {
-                return userId !== this.myUserId;
+        if (Array.isArray(heroes) && heroes.length > 0) {
+            // filter out ourselves just in case
+            this.heroes = heroes.filter((h) => {
+                return h.userId != this.myUserId;
             });
         }
 
-        this.emit(RoomEvent.Summary, summary);
+        summary["m.heroes"] = heroes?.map((h) => h.userId);
```

Comment:
```suggestion
        // Convert back to the legacy heroes format for the `RoomEvent.Summary` event.
        summary["m.heroes"] = heroes?.map((h) => h.userId);
```

Reply:
can't help but feel that modifying the passed-in object in this way is bad karma that will surprise someone in future. Maybe we should document this behaviour in the doc-comment?